app/code/Magento/ConfigurableProduct/Plugin/SalesRule/Model/Rule/Condition/Product.php

```diff
@@ -0,0 +1,53 @@
+<?php declare(strict_types=1);
```

Comment:
Please, cover new code at least by the unit test. It would be great if scenario will be covered with integration test

Reply:
Done.
@sidolov, please check and rerun Travis CI: there is an error which is not related to the codebase.